example.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * This is a hypothetical implementation to understand
+ * what implementations may look like as we define
+ * a standard for video quality of experience metrics.
+ */
+
+// States
+var NOT_PLAYING = 0;
+var ATTEMPTING_TO_PLAY = 1;
+var PLAYING = 2;
+
+
+// Watch Time
+var watchTime = 0;
+var lastWatchTimeCheck;
+var watchTimeInterval;
+
+function updateWatchTime() {
+  watchTime = watchTime + (Date.now() - lastWatchTimeCheck);
+  lastWatchTimeCheck = Date.now();
+}
+
+player.addEventListener('playrequest', function(){
+  lastWatchTimeCheck = Date.now();
+  watchTimeInterval = window.setInterval(updateWatchTime, 250);
+});
+
+function stopWatchTimeInterval() {
+  window.clearInterval(watchTimeInterval);
+  updateWatchTime();
+}
+
+player.addEventListener('pause', stopWatchTimeInterval);
+player.addEventListener('end', stopWatchTimeInterval);
+
+// Video Startup Time
+var videoStartupTime = null;
+var firstPlayRequestTime;
+
+player.addEventListener('playrequest', function(){
```

Comment:
Is the ideal here to only track the _first_ play request? This will override with every subsequent `playrequest`